runtime/lua/vim/lsp/handlers.lua

```diff
@@ -333,25 +333,25 @@ M['textDocument/implementation'] = location_handler
 ---     - border:     (default=nil)
 ---         - Add borders to the floating window
 ---         - See |vim.api.nvim_open_win()|
-function M.signature_help(_, method, result, _, bufnr, config)
+---     - parametersOnly: (default=false)
+---         - Display only a list of parameters
+function M.signature_help(_, method, result, _, _, config)
   config = config or {}
   -- When use `autocmd CompleteDone <silent><buffer> lua vim.lsp.buf.signature_help()` to call signatureHelp handler
   -- If the completion item doesn't have signatures It will make noise. Change to use `print` that can use `<silent>` to ignore
   if not (result and result.signatures and result.signatures[1]) then
     print('No signature help available')
     return
   end
-  local lines = util.convert_signature_help_to_markdown_lines(result)
+  local lines = util.convert_signature_help_to_markdown_lines(result, config.parametersOnly or false)
   lines = util.trim_empty_lines(lines)
   if vim.tbl_isempty(lines) then
     print('No signature help available')
     return
   end
-  local syntax = api.nvim_buf_get_option(bufnr, 'syntax')
-  local p_bufnr, _ = util.focusable_preview(method, function()
-    return lines, util.try_trim_markdown_code_blocks(lines), config
+  util.focusable_preview(method, function()
+    return lines, "markdown", config
   end)
-  api.nvim_buf_set_option(p_bufnr, 'syntax', syntax)
```

Comment:
This is going to remove syntax highlighting, which was a feature request that I closed.

Reply:
@mjlbach  I reverted this change. Will experiment with `nvim_buf_set_extmark`, of perhaps if you can explain what a custom highlight group is and how it can be used